datumaro/cli/commands/download.py

```diff
@@ -0,0 +1,101 @@
+from ..util import MultilineFormatter
+from ..util.errors import CliException
+from ..util.project import generate_next_file_name
+
+
+def build_parser(parser_ctor=argparse.ArgumentParser):
+    builtin_writers = sorted(Environment().converters)
+    if TFDS_EXTRACTOR_AVAILABLE:
+        available_datasets = ", ".join(
+            f'tfds:{name}' for name in AVAILABLE_TFDS_DATASETS)
+    else:
+        available_datasets = "N/A (install TensorFlow Datasets " \
+            "to download datasets)"
+
+    parser = parser_ctor(
+        help="Download a publicly available dataset and save it "
+            "in a given format",
+        description="""
+        TBD.|n
+        |n
+        Supported datasets: {}|n
+        |n
+        Supported output formats: {}|n
+        |n
+        Examples: TBD
+        """.format(available_datasets, ', '.join(builtin_writers)),
+        formatter_class=MultilineFormatter)
+
+    parser.add_argument('-i', '--dataset-id', required=True,
+        help="Which dataset to download")
+    parser.add_argument('-f', '--output-format', required=True,
+        help="Output format")
+    parser.add_argument('-o', '--output-dir', dest='dst_dir',
+        help="Directory to save output (default: a subdir in the current one)")
+    parser.add_argument('--overwrite', action='store_true',
+        help="Overwrite existing files in the save directory")
+    parser.add_argument('extra_args', nargs=argparse.REMAINDER,
+        help="Additional arguments for output format (pass '-- -h' for help). "
+            "Must be specified after the main command arguments")
+    parser.set_defaults(command=download_command)
+
+    return parser
+
+def get_sensitive_args():
+    return {
+        download_command: ['dst_dir', 'extra_args'],
+    }
+
+def download_command(args):
+    env = Environment()
+
+    try:
+        converter = env.converters[args.output_format]
+    except KeyError:
+        raise CliException("Converter for format '%s' is not found" %
+            args.output_format)
+    extra_args = converter.parse_cmdline(args.extra_args)
+
+    dst_dir = args.dst_dir
+    if dst_dir:
+        if not args.overwrite and osp.isdir(dst_dir) and os.listdir(dst_dir):
+            raise CliException("Directory '%s' already exists "
+                "(pass --overwrite to overwrite)" % dst_dir)
+    else:
+        dst_dir = generate_next_file_name('%s-%s' % (
+            make_file_name(args.dataset_id),
+            make_file_name(args.output_format),
+        ))
+    dst_dir = osp.abspath(dst_dir)
+
+    log.info("Downloading the dataset")
+    if TFDS_EXTRACTOR_AVAILABLE and args.dataset_id.startswith('tfds:'):
+        tfds_ds_name = args.dataset_id[5:]
+        if tfds_ds_name in AVAILABLE_TFDS_DATASETS:
+            extractor = make_tfds_extractor(tfds_ds_name)
+        else:
+            raise CliException(f"Unsupported TFDS dataset '{tfds_ds_name}'")
+    else:
+        raise CliException(f"Unknown dataset ID '{args.dataset_id}'")
```

Comment:
I think, in this case we should display plugin installation instructions.

Reply:
What plugin?

Reply:
`pip install datumaro[tfds]`

Reply:
Okay, I added that.